source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,220 @@
+
+  - ``description``: The name of the test.
+
+  - ``skipReason``: Optional, string describing why this test should be
+    skipped.
+
+  - ``clientOptions``: Optional, parameters to pass to MongoClient().
+
+    - ``clientSideEncryptionOptions``
+
+      - ``kmsProvider``: Fixed as "aws-kms".
+
+      - ``kmsProviderOptions``: Optional.
+
+        - ``uri``: Optional. If set to "localhost:12345", use a local mock KMS service. (TODO: mock KMS not available yet.) 
+
+      - ``extraOptions``:
+
+      - ``schemas``: An optional array of schemas for local configuration.
+
+        - ``database``: The database name for the target collection.
+
+        - ``collection``: The name of the target collection.
+
+        - ``schema``: The schema document.
+
+      - ``useRemoteSchemas``: An optional boolean. If true, schemas are retrieved from the MongoDB server using listCollections.
+
+  - ``kmsPreCmd``: Optional, a command to send to the mock KMS server
+    before the test starts.
+
+  - ``kmsPostCmd``: Optional, a command to send to the mock KMS server
+    after the test ends.
+
+  - ``mockupcryptdPostCmd``: Optional, a command to send to the mock
+    mongocryptd after the test ends.
+
+  - ``operations``: Array of documents, each describing an operation to be
+    executed. Each document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: If present, indicates the target collection.
+
+      - ``database``: The database name for the target collection.
+
+      - ``collection``: The name of the target collection.
+
+    - ``collectionOptions``: Optional, parameters to pass to the Collection()
+      used for this operation.
+
+    - ``command_name``: Present only when ``name`` is "runCommand". The name
+      of the command to run. Required for languages that are unable preserve
+      the order keys in the "command" argument when parsing JSON/YAML.
+
+    - ``arguments``: Optional, the names and values of arguments.
+
+    - ``result``: The return value from the operation, if any. This field may
+      be a single document or an array of documents in the case of a
+      multi-document read. If the operation is expected to return an error, the
+      ``result`` is a single document that has one or more of the following
+      fields:
+
+      - ``errorContains``: A substring of the expected error message.
+
+      - ``errorCodeName``: The expected "codeName" field in the server
+        error response.
+
+  - ``expectations``: Optional list of a subset of command monitoring events
+    that the driver must generate.
+
+  - ``outcome``: Array describing the return value and/or expected state of
+    the collection after the operation is executed. Contains the following
+    fields:
+
+    - ``database``: The database name for the target collection.
+
+    - ``collection``: The name of the target collection.
+
+    - ``data``: Expected data in the target collection.
```

Comment:
I think this syntax also deviates from other spec formats (see: [CRUD](https://github.com/mongodb/specifications/tree/master/source/crud/tests/README.rst) and [Transactions](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst)).

`data` would normally be nested under `collection`, which also supports a `name` property. Will FLE require a custom database name to be specified? If so, perhaps we can add `database_name` under `collection` as well. That would introduce new syntax, but not conflict with any prior art.